src/main/java/de/tum/in/www1/artemis/domain/notification/GroupNotificationFactory.java

```diff
@@ -84,6 +84,10 @@ public static GroupNotification createNotification(Exercise exercise, User autho
                 title = "Duplicate test case was found.";
                 text = "Exercise \"" + exercise.getTitle() + "\" has multiple test cases with the same name.";
             }
+            case ILLEGAL_SUBMISSION -> {
+                title = "Invalid submission of a student.";
+                text = "Exercise \"" + exercise.getTitle() + "\" has invalid submissions of students.";
```

Comment:
```suggestion
                text = "Exercise \"" + exercise.getTitle() + "\" received an illegal submission of a student.";
```

Reply:
please include the student name

Reply:
`author` does not seem to be used in the method, so we could rename it to affectedUser and add the login in the text

Reply:
`author` is used here: 

```
        GroupNotification notification = new GroupNotification(exercise.getCourseViaExerciseGroupOrCourseMember(), title, text, author, groupNotificationType);
```

Reply:
But the notification text is created in the `notifyPush()` method. This is just a dummy text.

Reply:
We could conditionally on a ILLEGAL_SUBMISSION use the author for our purpose and set it to null for the notification author